core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -480,15 +572,39 @@ public boolean token(String currentFieldName, XContentParser.Token token, XConte
         public IncludeExclude createIncludeExclude(Map<ParseField, Object> otherOptions) {
             Object includeObject = otherOptions.get(INCLUDE_FIELD);
             String include = null;
+            int partition = -1;
+            int pOf =-1;
             SortedSet<BytesRef> includeValues = null;
             if (includeObject != null) {
                 if (includeObject instanceof String) {
                     include = (String) includeObject;
                 } else if (includeObject instanceof SortedSet) {
                     includeValues = (SortedSet<BytesRef>) includeObject;
+                } else if (includeObject instanceof Integer) {
+                    partition = (Integer) includeObject;
+                    Object ofObject = otherOptions.get(OF_FIELD);
+                    if(ofObject instanceof Integer){
+                        pOf = (Integer) ofObject;
+                        if (pOf < 2){
+                            throw new IllegalArgumentException(PARTITION_FIELD.getPreferredName()+" must be >1");                                                    
```

Comment:
OK - I'll change`of` to be `num_partitions` just because "19 _of_ 20" seems misleading